File: March 2024 GFG SOLUTION/March-23.cpp
//{ Driver Code Starts
import java.io.*;
import java.util.*;

class Main {
    // Driver code
    public static void main(String[] args) throws Exception {
        BufferedReader br =
            new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine().trim());
        while (t-- > 0) {
            int n = Integer.parseInt(br.readLine().trim());
            Solution obj = new Solution();
            int ans[] = obj.Series(n);
            for (int i : ans) {
                System.out.print(i + " ");
            }
            System.out.println();
        }
    }
}
// } Driver Code Ends


class Solution {
    int[] Series(int n) {
        int MOD = 1000000007;
        int[] ans = new int[n + 1];
        if (n == 0) return ans; 
        
        int prev = 0, cur = 1;
        ans[0] = prev;
        ans[1] = cur;
        
        for (int i = 2; i <= n; ++i) {
            int next = (prev + cur) % MOD;
            ans[i] = next;
            prev = cur;
            cur = next;
        }
        
        return ans;
    }

}

File: March 2024 GFG SOLUTION/March-23.py
#User function Template for python3

class Solution:
    def series(self, n):
        MOD = 10**9 + 7
        ans = []
        if n == 0:
            return ans  # Handle edge case
        
        prev, cur = 0, 1
        ans.append(prev)
        ans.append(cur)
        
        for i in range(2, n + 1):
            next_val = (prev + cur) % MOD
            ans.append(next_val)
            prev, cur = cur, next_val
        
        return ans


#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    t = int(input())
    for _ in range(t):
        N = int(input())
        ob = Solution()
        result = ob.series(N)
        print(*result)
# } Driver Code Ends
